pandas/sparse/array.py

```diff
@@ -769,14 +769,17 @@ def make_sparse(arr, kind='block', fill_value=None):
     if isnull(fill_value):
         mask = notnull(arr)
     else:
-        mask = arr != fill_value
+        mask = np.not_equal(arr, fill_value)
+        # In NumPy 1.12.0, not implemented at least for arr.dtype=str
+        if mask is NotImplemented:
+            mask = np.not_equal(arr.astype(object), fill_value)
```

Comment:
FYI #15587 is going to validate fill values (but of course let's fix here for what you need).

``isinstance(fill_value, compat.string_types)`` is idiom for checking string-likes (if its a scalar)